### PR TITLE
CTX-5524: Created template for compiling onnx models to zkml format

### PR DESCRIPTION
**Pull request by VukManojlovic**

### CODE REVIEW

File: tasks/zkml-convert-model/main.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+import json
+import asyncio
+import logging
+
+from coretex import currentTaskRun, folder_manager, Model
+
+import ezkl
+import onnx
+import numpy as np
```
Comment (bogdant36) [inline tasks/zkml-convert-model/main.py:11]: sort imports correctly

File: tasks/zkml-convert-model/main.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+
+def linkFolder(source: Path, destination: Path) -> None:
+    destination.mkdir(exist_ok = True)
+    for item in source.rglob('*'):
```
Comment (bogdant36) [inline tasks/zkml-convert-model/main.py:16]: use "" instead of ''

File: tasks/zkml-convert-model/main.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        destItem = destination / item.relative_to(source)
+
+        if item.is_dir():
+            continue
+        elif item.is_file():
+            destItem.parent.mkdir(parents=True, exist_ok=True)
```
Comment (bogdant36) [inline tasks/zkml-convert-model/main.py:22]: format properly whitespace around "="

File: tasks/zkml-convert-model/main.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+
+def linkFolder(source: Path, destination: Path) -> None:
+    destination.mkdir(exist_ok = True)
+    for item in source.rglob("*"):
+        destItem = destination / item.relative_to(source)
+
+        if item.is_dir():
+            continue
+        elif item.is_file():
```
Comment (dule1322) [inline tasks/zkml-convert-model/main.py:21]: `elif` can be removed as the `if` is already doing `continue` if the criteria is met

File: tasks/zkml-convert-model/task.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- name: outputs
+  params:
+  - name: outputCircuit
+    description: The dataset to which this TaskRun will output it's results
+    value: null
+    data_type: model
+    required: false
+- name: parameters
+  params:
+  - name: inputVisibility
```
Comment (dule1322) [inline tasks/zkml-convert-model/task.yaml:22]: I'd rename this to `privateInput` or `publicInput`. Can you do this for other 2 visibility params as well?

File: tasks/zkml-convert-model/main.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    logging.info(">> [ZKML] Generating settings")
+    ezkl.gen_settings(onnxPath, settingsPath, py_run_args = pyRunArgs)
+
+    logging.info(">> [ZKML] Compiling model")
+    asyncio.run(compileModel(onnxPath, settingsPath, compiledModelPath, verifKey, proofKey))
+
+    logging.info(">> [ZKML] EZKL setup complete")
+    logging.info(">> [ZKML] Uploading model and EZKL files")
+    ctxModel = Model.createModel(taskRun.generateEntityName(), taskRun.projectId, ctxOnnxModel.accuracy, {})
+    ctxModel.upload(modelDir)
```
Comment (dule1322) [inline tasks/zkml-convert-model/main.py:117]: Tag the model and you didn't submit it as output of this task as well.

**Review by dule1322 — CHANGES_REQUESTED** [review]

**Review by dule1322 — APPROVED** [review]

**Pull request merged**